src/spatial_ref/srs.rs

```diff
@@ -226,41 +226,41 @@ impl SpatialRef {
                 method_name: "OSRExportToPROJJSON",
             })
         } else {
-            Ok(_string(c_projjsonstr))
+            Ok(_string(c_projjsonstr).unwrap_or_default())
         };
         unsafe { gdal_sys::VSIFree(c_projjsonstr.cast::<std::ffi::c_void>()) };
         res
     }
 
-    pub fn auth_name(&self) -> Result<String> {
+    pub fn auth_name(&self) -> Option<String> {
         let c_ptr = unsafe { gdal_sys::OSRGetAuthorityName(self.0, ptr::null()) };
```

Comment:
The same is assumed to be owned by `self` so there is no need to free `c_ptr`?

Reply:
Owned by PROJ via `OGRSpatialReference`.